package.json

```diff
@@ -11,14 +11,21 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom",
+    "lint": "eslint --ext .js,.jsx ./",
+    "test": "CI=true react-scripts test --env=jsdom",
+    "test:watch": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject",
     "storybook": "start-storybook -c src/components/.storybook -p 4000",
-    "build-storybook": "build-storybook"
+    "storybook:build": "build-storybook"
   },
+  "precommit": ["lint", "test"],
```

Comment:
new `pre-commit` package runs these npm scripts in order BEFORE any code can be committed. that was really easy to set up.